src/utilities/mergeParams.spec.ts

```diff
@@ -0,0 +1,55 @@
+import { describe, expect, test } from 'vitest'
+import { Param } from '@/types'
+import { mergeParams } from '@/utilities'
+
+describe('mergeParams', () => {
+  test('given params without overlap, returns combination of both', () => {
+    const input: Record<string, Param[]>[] = [
+      {
+        foo: [String],
+        bar: [String],
+      },
+      {
+        zoo: [String],
+      },
+    ]
+
+    const response = mergeParams(...input)
+
+    expect(response).toHaveProperty('foo')
+    expect(response).toHaveProperty('bar')
+    expect(response).toHaveProperty('zoo')
+  })
+
+  test('given params with overlap of same type, does not combine into new tuple', () => {
+    const input: Record<string, Param[]>[] = [
+      {
+        foo: [String],
+      },
+      {
+        foo: [String],
+      },
+    ]
+
+    const response = mergeParams(...input)
+
+    expect(response).toHaveProperty('foo')
+    expect(response.foo).toMatchObject([String])
+  })
```

Comment:
This seems incorrect to me. Shouldn't this type be `[String, String]`? The tuple is a direct representation of how many params by that name there are in the order the appear.

Reply:
agreed. I think this was originally written to solve a different problem and never got revisited. Good catch